lib/api/api-connect.js

```diff
@@ -50,7 +50,13 @@ class ConnectHandler extends AsyncResource {
     removeSignal(this)
 
     this.callback = null
-    const headers = this.responseHeaders === 'raw' ? util.parseRawHeaders(rawHeaders) : util.parseHeaders(rawHeaders)
+
+    let headers
+    // Indicates is an HTTP2Session
+    if (headers != null) {
```

Comment:
Isn't this always true?

Reply:
Yes. For two reasons; there's no `upgrade` on H2, and when using `Client#connect`, the intention is to return the H2 Stream, as soon as it has an identifier assigned.

As we do not wait for the response, but rather just there's an `identifier` assigned, we do not read the `headers` at this point but rather is up to the implementation that has control of the stream to decide what to do with it.

Reply:
What's the point of the condition then?

Reply:
Handle the requests for `http/1` as well

Reply:
the issue is that header is always undefined there because its not being assigned to anything

Reply:
Ah... My bad, sorry, I read it wrong 😓 
It is `rawHeaders` the one that should be checked.